transaction/src/signing/ecdsa_secp256k1.rs

```diff
@@ -11,9 +11,8 @@ impl EcdsaSecp256k1PrivateKey {
         EcdsaSecp256k1PublicKey(PublicKey::from_secret_key_global(&self.0).serialize())
     }
 
-    pub fn sign(&self, msg: &[u8]) -> EcdsaSecp256k1Signature {
-        let h = sha256(sha256(msg));
-        let m = Message::from_slice(&h.0).expect("Hash is always a valid message");
+    pub fn sign(&self, msg_hash: &[u8]) -> EcdsaSecp256k1Signature {
```

Comment:
Change interface to `message_hash: &Hash` instead?

Reply:
To be consistent with the interface of `EddsaEd25519PrivateKey` I would recommend that the passed argument is the message and not the message hash.

Reply:
> To be consistent with the interface of EddsaEd25519PrivateKey I would recommend that the passed argument is the message and not the message hash.
Actually, it should be message hash in `EcdsaSecp256k1Signature`.  Thanks for spotting this @0xOmarA